### PR TITLE
feat: update to angular 12 

### PR DESCRIPTION
**Pull request by jafin**
* Update component to Angular 12
* Replace TSLint with ESLint

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -44,3 +44,4 @@ testem.log
 Thumbs.db
 
 TESTS-*.xml
+/angular-reactive-validation/node_modules/*
```
Comment (davidwalschots) [inline .gitignore]: **thought**: Probably you ran `npm install` within `/angular-reactive-validation`. There's no need to do so. Run `npm install` in the root and remove the `node_modules` folder in the subfolder. Everything should still work and this line can be removed again 😀.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
+{
```
Comment (davidwalschots) [inline .eslintrc.json:1]: Thanks for migrating to eslint. I'd like to note that on opening various ts files within vscode I'm seeing a lot of eslint errors popping up.
**suggestion**: Go through each individual ts file and apply the rules.

File: angular-reactive-validation/package.json [hunk 3]
```diff
@@ -20,18 +20,18 @@
...
   "dependencies": {},
   "peerDependencies": {
-    "@angular/core": "^11.0.0",
-    "@angular/common": "^11.0.0",
-    "@angular/forms": "^11.0.0",
-    "rxjs": "^6.5.4"
+    "@angular/core": "^12.2.15",
+    "@angular/common": "^12.2.15",
+    "@angular/forms": "^12.2.15",
+    "rxjs": "^6.6.7"
```
Comment (davidwalschots) [inline angular-reactive-validation/package.json]: **thought**: The package can be compatible with any Angular 12.x version by changing the `peerDependencies` to be the lowest possible version we want to be compatible with. In this case `"^12.0.0"` for Angular and `"^6.5.3"` for rxjs (so can actually be lower than the previous version set there, oops).

File: angular-reactive-validation/package-lock.json [hunk 4]
```diff
@@ -0,0 +1,252 @@
+{
```
Comment (davidwalschots) [inline angular-reactive-validation/package-lock.json]: As noted earlier, you probably ran `npm install` in the sub directory, causing this file to be created.
**suggestion**: Remove the file.

File: src/app/app.component.html [hunk 5]
```diff
@@ -1,27 +1,44 @@
...
+          #minlengthValidation
+          for="firstName"
+          key="minlength"
+        >
+          minLength validation test!!!
+          {{ minlengthValidation.context?.requiredLength }}
+        </arv-validation-message>
+      </arv-validation-messages>
+    </div>
+    <div>
```
Comment (davidwalschots) [inline src/app/app.component.html:32]: Textual alignment of this div and some of its content is a bit wonky. 😀

**Review by davidwalschots — CHANGES_REQUESTED** [review]
Thank you for your contribution! I have taken the liberty to modify the node version the build pipeline uses to v14. Some small remarks here and there. Looking forward to getting this merged!

Do you by any chance plan on updating to Angular 13 right after this PR?

File: .gitignore (hunk 1 above)
Comment (jafin) [reply .gitignore]: Yes i did. Fixed.

File: .eslintrc.json (hunk 2 above)
Comment (jafin) [reply .eslintrc.json:1]: Ok fixed, 1) I didn't see this I was in Webstorm. and 2) running `ng lint` did not pick up any lint warnings.. Perhaps something is still missing as I would of expected ng lint to display warnings you were seeing?

File: angular-reactive-validation/package-lock.json (hunk 4 above)
Comment (jafin) [reply angular-reactive-validation/package-lock.json]: Fixed. removed.

File: angular-reactive-validation/package.json (hunk 3 above)
Comment (jafin) [reply angular-reactive-validation/package.json]: done

**Comment by jafin** [comment]
> Do you by any chance plan on updating to Angular 13 right after this PR?

No actual plan to go to v13 at the moment for me.

File: .eslintrc.json (hunk 2 above)
Comment (davidwalschots) [reply .eslintrc.json:1]: I've included additional paths for linting in `angular.json`. It feels somewhat odd to have to define more paths while within the `"projects" -> "angular-reactive-validation"` scope. That could be related to the fact that when I introduced the project in a project, there was no true angular support for doing so yet.

**Comment by jafin** [comment]
@davidwalschots thanks for fixing up the PR!

**Comment by davidwalschots** [comment]
Thanks for your contribution!

**Pull request merged**